packages/basic-crawler/src/internals/basic-crawler.ts

```diff
@@ -943,7 +943,7 @@ export class BasicCrawler<Context extends CrawlingContext = BasicCrawlingContext
 
             // reclaim session if request finishes successfully
             request.state = RequestState.DONE;
-            session?.markGood();
+            crawlingContext.session?.markGood();
```

Comment:
I'm not seeing how this helps since both `session` and `crawlingContext.session` are pointing out to the same reference. Or is there some issue with the session initialization?

Reply:
Makes sense. Indeed, those mutations...